src/parcsr_mv/par_csr_matop.c

```diff
@@ -5048,79 +5053,53 @@ hypre_ParcsrAdd( HYPRE_Complex        alpha,
-   HYPRE_Int         nnz_offd_A = hypre_CSRMatrixNumNonzeros(A_offd);
    HYPRE_BigInt *col_map_offd_A = hypre_ParCSRMatrixColMapOffd(A);
    HYPRE_Int          *A2C_offd;
 
    /* diag part of B */
    hypre_CSRMatrix    *B_diag   = hypre_ParCSRMatrixDiag(B);
-   HYPRE_Complex      *B_diag_a = hypre_CSRMatrixData(B_diag);
-   HYPRE_Int          *B_diag_i = hypre_CSRMatrixI(B_diag);
-   HYPRE_Int          *B_diag_j = hypre_CSRMatrixJ(B_diag);
    HYPRE_Int     *rownnz_diag_B = hypre_CSRMatrixRownnz(B_diag);
    HYPRE_Int  num_rownnz_diag_B = hypre_CSRMatrixNumRownnz(B_diag);
    HYPRE_Int    num_rows_diag_B = hypre_CSRMatrixNumRows(B_diag);
    HYPRE_Int    num_cols_diag_B = hypre_CSRMatrixNumCols(B_diag);
-   HYPRE_Int         nnz_diag_B = hypre_CSRMatrixNumNonzeros(B_diag);
 
    /* off-diag part of B */
    hypre_CSRMatrix    *B_offd   = hypre_ParCSRMatrixOffd(B);
-   HYPRE_Complex      *B_offd_a = hypre_CSRMatrixData(B_offd);
-   HYPRE_Int          *B_offd_i = hypre_CSRMatrixI(B_offd);
-   HYPRE_Int          *B_offd_j = hypre_CSRMatrixJ(B_offd);
    HYPRE_Int     *rownnz_offd_B = hypre_CSRMatrixRownnz(B_offd);
    HYPRE_Int  num_rownnz_offd_B = hypre_CSRMatrixNumRownnz(B_offd);
    HYPRE_Int    num_rows_offd_B = hypre_CSRMatrixNumRows(B_offd);
    HYPRE_Int    num_cols_offd_B = hypre_CSRMatrixNumCols(B_offd);
-   HYPRE_Int         nnz_offd_B = hypre_CSRMatrixNumNonzeros(B_offd);
    HYPRE_BigInt *col_map_offd_B = hypre_ParCSRMatrixColMapOffd(B);
    HYPRE_Int          *B2C_offd;
 
-   /* C */
+   /* C data */
    hypre_ParCSRMatrix   *C;
    HYPRE_BigInt         *row_starts_C;
    HYPRE_BigInt         *col_starts_C;
    hypre_CSRMatrix      *C_diag;
    hypre_CSRMatrix      *C_offd;
-
-   HYPRE_Int             num_cols_offd_C = num_cols_offd_A + num_cols_offd_B;
    HYPRE_BigInt         *col_map_offd_C;
-
-   HYPRE_Int             nnz_diag_C_alloc = nnz_diag_A + nnz_diag_B;
-   HYPRE_Int             nnz_offd_C_alloc = nnz_offd_A + nnz_offd_B;
-   HYPRE_Int             nnz_diag_C = 0;
-   HYPRE_Int             nnz_offd_C = 0;
-
    HYPRE_Int            *C_diag_i, *C_offd_i;
-   HYPRE_Int            *C_diag_j, *C_offd_j;
-   HYPRE_Complex        *C_diag_a, *C_offd_a;
-   HYPRE_Int            *marker_diag, *marker_offd;
-
-   HYPRE_Int             num_rownnz_diag_C;
-   HYPRE_Int             num_rownnz_offd_C;
    HYPRE_Int            *rownnz_diag_C = NULL;
    HYPRE_Int            *rownnz_offd_C = NULL;
-
-   HYPRE_Int             num_procs, my_id;
-   HYPRE_Int             i, ii, j;
+   HYPRE_Int             num_rownnz_diag_C;
+   HYPRE_Int             num_rownnz_offd_C;
+   HYPRE_Int             num_rows_diag_C = num_rows_diag_A;
+   HYPRE_Int             num_cols_diag_C = num_cols_diag_A;
+   HYPRE_Int             num_rows_offd_C = num_rows_offd_A;
+   HYPRE_Int             num_cols_offd_C = num_cols_offd_A + num_cols_offd_B;
+   HYPRE_Int             twspace[hypre_NumThreads()];
```

Comment:
may use `malloc` rather than variable length array

Reply:
The use of `malloc` is preferred since VLA is not in C++ standard, right (just checking)? This is used here as well: https://github.com/hypre-space/hypre/blob/3f12d47651ecf18644d95bc9e975c572fe44ccf3/src/parcsr_ls/aux_interp.c#L683

I'll change these to malloc

Reply:
Not familiar with VLA in C/C++. I just found out it is indeed in C99. (I thought this sort of allocation is not allowed in C...). So, I don't have strong options on it. Maybe other people can comment on it.

Reply:
Ok! I decided to change to malloc anyway. If we ever decide to use a C++ compiler in these files, it will not create problems. And this is a very small allocation happening very few times, so there aren't benefits of using stack over heap memory

Reply:
I agree with changing to malloc anyway.  Thanks!

Reply:
Good! Thanks, Rob!